src/test/scala/io/iohk/ethereum/network/rlpx/RLPxConnectionHandlerSpec.scala

```diff
@@ -70,6 +73,23 @@ class RLPxConnectionHandlerSpec extends FlatSpec with Matchers with MockFactory
     connection.expectNoMsg()
   }
 
+  it should "close the connection when Ack timeout happens" in new TestSetup {
+    (mockMessageCodec.encodeMessage _).expects(Ping(): MessageSerializable).returning(ByteString("ping encoded")).anyNumberOfTimes()
+
+    setupRLPxConnection()
+
+    rlpxConnectionParent watch rlpxConnection
+
+    rlpxConnection ! RLPxConnectionHandler.SendMessage(Ping())
+    connection.expectMsg(Tcp.Write(ByteString("ping encoded"), RLPxConnectionHandler.Ack))
+
+    //Wait for Ack timeout
+    Thread.sleep(rlpxConfiguration.waitForTcpAckTimeout.toMillis)
```

Comment:
Added using `max` instead of `Thread.sleep` :+1: I used: `rlpxConnectionParent.expectTerminated(rlpxConnection, max = 2 * rlpxConfiguration.waitForTcpAckTimeout)` as the test was sometimes failing as it the RLPxConnecitonHandler didn't have enough time to process the timeout.